sw/device/tests/entropy_src_fw_observe_many_contiguous.c

```diff
@@ -0,0 +1,170 @@
+  dif_entropy_src_t *entropy_src,
+  uint32_t nr_samples,
+  uint32_t timeout_usec,
+  uint32_t repeat_count) {
+  CHECK_DIF_OK(dif_entropy_src_set_enabled(entropy_src, kDifToggleDisabled));
+
+  // Disable all health tests.
+  CHECK_DIF_OK(dif_entropy_src_health_test_configure(
+      entropy_src, (dif_entropy_src_health_test_config_t){
+                        .test_type = kDifEntropySrcTestRepetitionCount,
+                        .high_threshold = 0xffffffff,
+                        .low_threshold = 0}));
+  CHECK_DIF_OK(dif_entropy_src_health_test_configure(
+      entropy_src, (dif_entropy_src_health_test_config_t){
+                        .test_type = kDifEntropySrcTestRepetitionCountSymbol,
+                        .high_threshold = 0xffffffff,
+                        .low_threshold = 0}));
+  CHECK_DIF_OK(dif_entropy_src_health_test_configure(
+      entropy_src,
+      (dif_entropy_src_health_test_config_t){
+          .test_type = kDifEntropySrcTestAdaptiveProportion,
+          .high_threshold =
+              0xffffffff,
+          .low_threshold =
+              0}));
+  CHECK_DIF_OK(dif_entropy_src_health_test_configure(
+      entropy_src, (dif_entropy_src_health_test_config_t){
+                        .test_type = kDifEntropySrcTestBucket,
+                        .high_threshold = 0xffffffff,
+                        .low_threshold = 0}));
+  CHECK_DIF_OK(dif_entropy_src_health_test_configure(
+      entropy_src, (dif_entropy_src_health_test_config_t){
+                        .test_type = kDifEntropySrcTestMarkov,
+                        .high_threshold = 0xffffffff,
+                        .low_threshold = 0}));
+  CHECK_DIF_OK(dif_entropy_src_fw_override_configure(
+      entropy_src,
+      (dif_entropy_src_fw_override_config_t){
+          .entropy_insert_enable = false,
+          .buffer_threshold = kEntropySrcFifoThreshold,
+      },
+      kDifToggleEnabled));
+  CHECK_DIF_OK(dif_entropy_src_configure(
+      entropy_src,
+      (dif_entropy_src_config_t){
+          .fips_enable = true,
+          .route_to_firmware = false,
+          .bypass_conditioner = false,
+          .single_bit_mode = kDifEntropySrcSingleBitModeDisabled,
+          .health_test_threshold_scope = false,
+          .health_test_window_size = kEntropySrcHealthTestWindowSize,
+          .alert_threshold = UINT16_MAX},
+      kDifToggleEnabled));
+
+  while (repeat_count-- > 0) {
+    LOG_INFO("collecting %u samples...", nr_samples);
+    // Collect samples from the the observe FIFO.
+    uint32_t words_to_read = ceil_div(nr_samples * 4, sizeof(uint32_t));
+    uint32_t *sample_buffer_ptr = sample_buffer;
+    ibex_timeout_t tmo = ibex_timeout_init(timeout_usec);
+    while(words_to_read > 0 && !ibex_timeout_check(&tmo)) {
+      uint32_t rem = words_to_read;
+      if (rem > kEntropySrcFifoThreshold) {
+        rem = kEntropySrcFifoThreshold;
+      }
+      CHECK_DIF_OK(dif_entropy_src_observe_fifo_blocking_read(
```

Comment:
Maybe here / before this call we could read the FIFO_DEPTH to track the max fill level over time.